sdk/src/lib.rs

```diff
@@ -110,6 +110,11 @@ impl ProverClient {
             .as_ref()
             .ok_or_else(|| anyhow::anyhow!("Network client not initialized"))?;
 
+        // Execute the runtime before creating the proof request.
+        let mut runtime = Runtime::new(Program::from(elf));
+        runtime.write_vecs(&stdin.buffer);
+        runtime.run();
+
```

Comment:
could be overkill, but an optional `skip_simulation` bool that is false by default might be nice

Reply:
Will skip this for now, left a TODO about how we can add this for later if there's demand.